src/Juvix/Compiler/Internal/Data/InstanceInfo.hs

```diff
@@ -29,20 +29,34 @@ data InstanceInfo = InstanceInfo
   }
 
 -- | Maps trait names to available instances
-type InstanceTable = HashMap InductiveName [InstanceInfo]
+newtype InstanceTable = InstanceTable
+  { _instanceTableMap :: HashMap InductiveName [InstanceInfo]
+  }
 
 makeLenses ''InstanceApp
 makeLenses ''InstanceInfo
+makeLenses ''InstanceTable
+
+instance Semigroup InstanceTable where
+  t1 <> t2 =
+    InstanceTable $
+      HashMap.unionWith (++) (t1 ^. instanceTableMap) (t2 ^. instanceTableMap)
```

Comment:
Should we handle deletion of duplicates or is it guaranteed that these two lists are disjoint?

Reply:
I think they will be disjoint - this is invoked when we're combining things from different modules

Reply:
If I open the same module twice I'm getting an ambiguity error.

`Main.juvix`
```
module Main;

import Default open;
import Default open;

x : Unit := default;
```

Default.jvuix
```
module Default;

type Unit := unit;

trait
type Default A := mkDefault { default : A };

instance
Unit-Default : Default Unit := mkDefault unit;

default {A : Type} {{x : Default A}} : A := case x
 | mkDefault a := a;
```

Reply:
Ah, Ok, then I'll add filtering duplicates

Reply:
done

Reply:
did you push the changes? I can't see it in the diff